healthcareai/supervised_model_trainer.py

```diff
@@ -47,6 +47,26 @@ def __init__(self, dataframe, predicted_column, model_type, impute=True, grain_c
         # Split the data into train and test
         self._advanced_trainer.train_test_split()
 
+        # Save the original dataframe columns to the parent class
+        self._advanced_trainer.pre_dummified_columns = dataframe.columns.values
```

Comment:
I changed the name of pre_dummified_columns to colnames_pre_pipeline and passed it as an optional argument to the constructor.  The name and documentation should make it clear when to use this argument.